src/lib/merkle_mask/masking_merkle_tree.ml

```diff
@@ -493,7 +495,6 @@ module Make (Inputs : Inputs_intf.S) = struct
           Base.merkle_root ancestor
 
     let remove_account_and_update_hashes t location =
-      t.accumulated <- None ;
```

Comment:
My original thought was that you could just remove this line here. No other changes should be required.

Among all the children of the root breadcrumb. If it's not on the canonical chain, then they would be treated as garbage and those would be collected, so I think there's no need to manually set `accumulated` to be None. For the mask on the canonical chain, we don't need to do anything. So remove this would be enough.

What you did here adds an extra bit of complexity to the code and also do the cleanup manually. I am not sure whether this is the best choice.